sygra/core/eval/metrics/aggregator_metrics/f1_score.py

```diff
@@ -0,0 +1,86 @@
+"""
+F1 Score Metric
+
+F1 = 2 * (Precision * Recall) / (Precision + Recall)
+Harmonic mean of precision and recall.
+"""
+
+from typing import Any, Dict, List
+
+from sygra.core.eval.metrics.aggregator_metrics.aggregator_metric_registry import aggregator_metric
+from sygra.core.eval.metrics.aggregator_metrics.base_aggregator_metric import BaseAggregatorMetric
+from sygra.core.eval.metrics.aggregator_metrics.precision import PrecisionMetric
+from sygra.core.eval.metrics.aggregator_metrics.recall import RecallMetric
+from sygra.core.eval.metrics.unit_metrics.unit_metric_result import UnitMetricResult
+from sygra.logger.logger_config import logger
+
+
+@aggregator_metric("f1_score")
+class F1ScoreMetric(BaseAggregatorMetric):
+    """
+    Generic F1 score metric.
+
+    Caller specifies positive class via keys and values.
+    """
+
+    def __init__(self, predicted_key: str, golden_key: str, positive_class: Any):
+        """
+        Args:
+            predicted_key: Key in predicted dict (e.g., "tool", "class")
+                          This is a required parameter to ensure explicit configuration.
+            golden_key: Key in golden dict (e.g., "event", "class")
+                       This is a required parameter to ensure explicit configuration.
+            positive_class: Value representing positive class (e.g., "click", 1, True)
+                          This is a required parameter since F1 is a class-wise metric.
+                          Without it, the metric would be equivalent to accuracy.
+
+        Raises:
+            ValueError: If predicted_key/golden_key is empty or positive_class is None
+        """
+        if not predicted_key:
+            raise ValueError("F1ScoreMetric: predicted_key cannot be empty")
+        if not golden_key:
+            raise ValueError("F1ScoreMetric: golden_key cannot be empty")
+        if positive_class is None:
+            raise ValueError("F1ScoreMetric: positive_class is required (cannot be None)")
```

Comment:
We can do the following to better keep auto consistency between the class names and the logging statements, in case the class name gets changed in the future. 

```suggestion
            raise ValueError(f"{self.__class__.__name__}: predicted_key cannot be empty")
        if not golden_key:
            raise ValueError(f"{self.__class__.__name__}: golden_key cannot be empty")
        if positive_class is None:
            raise ValueError(f"{self.__class__.__name__}: positive_class is required (cannot be None)")
```

Or, alternatively create a helper method/property in the base class the fetches the class name. 


```python
class BaseAggregatorMetric:
    @property
    def class_name(self) -> str:
        return self.__class__.__name__
```